hack/run-operator-catalog.sh

```diff
@@ -0,0 +1,108 @@
+# - KUBECTL_BIN: path to kubectl (or equivalent command line)
+# - SED_BIN: path to GNU sed
+# - CATALOG_NAMESPACE: Namespace to deploy the catalog. Defaults to shipwright-operator.
+# - SUBSCRIPTION_NAMESPACE: Namespace to install the operator via an OLM subscription. Defaults to
+#   shipwright-operator.
+# - NAME_PREFIX: prefix to use for all resource names. Defaults to "shipwright-"
+
+set -e
+
+KUSTOMIZE_BIN="${KUSTOMIZE_BIN:-${PWD}/bin/kustomize}"
+KUBECTL_BIN="${KUBECTL_BIN:-kubectl}"
+SED_BIN="${SED_BIN:-sed}"
+CATALOG_NAMESPACE="${CATALOG_NAMESPACE:-shipwright-operator}"
+SUBSCRIPTION_NAMESPACE="${SUBSCRIPTION_NAMESPACE:-shipwright-operator}"
+NAME_PREFIX="${NAME_PREFIX:-shipwright-}"
+
+if [[ -z ${CATALOG_IMG} ]]; then
+    echo "CATALOG_IMG environment variable must be set"
+    exit 1
+fi
+
+if [[ -z ${CSV_VERSION} ]]; then
+    echo "CSV_VERSION environment variable must be set"
+    exit 1
+fi
+
+function add_kustomizations() {
+    echo "Adding replacements not supported by kustomize"
+    ${SED_BIN} -i -E "s|image: (.+)$|image: ${CATALOG_IMG}|g" config/catalog/catalog_source.yaml
+    ${SED_BIN} -i -E "s|startingCSV: (.+)$|startingCSV: shipwright-operator.v${CSV_VERSION}|g" config/subscription/subscription.yaml
+    ${SED_BIN} -i -E "s|sourceNamespace: (.+)$|sourceNamespace: ${CATALOG_NAMESPACE}|g" config/subscription/subscription.yaml
+    ${SED_BIN} -i -E "s|source: (.+)$|source: ${NAME_PREFIX}operator|g" config/subscription/subscription.yaml
+
+    echo "Applying catalog source and subscription from kustomize"
+
+    pushd config/catalog
+    ${KUSTOMIZE_BIN} edit set namespace "${CATALOG_NAMESPACE}"
+    ${KUSTOMIZE_BIN} edit set nameprefix "${NAME_PREFIX}"
+    popd
+
+    pushd config/subscription
+    ${KUSTOMIZE_BIN} edit set namespace "${SUBSCRIPTION_NAMESPACE}"
+    ${KUSTOMIZE_BIN} edit set nameprefix "${NAME_PREFIX}"
+    popd
+}
+
+function dump_state() {
+    echo "Dumping OLM catalog sources"
+    ${KUBECTL_BIN} get catalogsources -n "${CATALOG_NAMESPACE}" -o yaml
+    echo "Dumping OLM subscriptions"
+    ${KUBECTL_BIN} get subscriptions -n "${SUBSCRIPTION_NAMESPACE}" -o yaml
+    echo "Dumping OLM installplans"
+    ${KUBECTL_BIN} get installplans -n "${SUBSCRIPTION_NAMESPACE}" -o yaml
+    echo "Dumping OLM CSVs"
+    ${KUBECTL_BIN} get clusterserviceversions -n "${SUBSCRIPTION_NAMESPACE}" -o yaml
+    echo "Dumping pods"
+    ${KUBECTL_BIN} get pods -n "${SUBSCRIPTION_NAMESPACE}" -o yaml
+    if [[ ${CATALOG_NAMESPACE} -ne ${SUBSCRIPTION_NAMESPACE} ]]; then
+        ${KUBECTL_BIN} get pods -n "${CATALOG_NAMESPACE}" -o yaml
+    fi
+}
+
+function wait_for_pod() {
+    label=$1
+    namespace=$2
+    timeout=$3
+    # Pod may not exist, in which case wait 30 seconds and try again
+    ${KUBECTL_BIN} wait --for=condition=Ready pod -l "${label}" -n "${namespace}" --timeout "${timeout}" || \
+        sleep 30 && ${KUBECTL_BIN} wait --for=condition=Ready pod -l "${label}" -n "${namespace}" --timeout "${timeout}"
```

Comment:
The 30 second gap before the retry is needed because OLM does a lot of work to deploy the operator, and no pods may exist when the initial wait call is made.